security.rst

```diff
@@ -511,149 +250,108 @@ is ``bcrypt``:
                 http://symfony.com/schema/dic/services/services-1.0.xsd">
 
             <config>
-                <!-- ... -->
-
-                <encoder class="Symfony\Component\Security\Core\User\User"
-                    algorithm="bcrypt"
-                    cost="12" />
+                <firewall name="dev"
+                    pattern="^/(_(profiler|wdt)|css|images|js)/"
+                    security="false" />
 
-                <!-- ... -->
+                <firewall name="main">
+                    <anonymous />
+                </firewall>
             </config>
         </srv:container>
 
     .. code-block:: php
 
         // config/packages/security.php
         $container->loadFromExtension('security', array(
-            // ...
-
-            'encoders' => array(
-                'Symfony\Component\Security\Core\User\User' => array(
-                    'algorithm' => 'bcrypt',
-                    'cost' => 12,
-                )
+            'firewalls' => array(
+                'dev' => array(
+                    'pattern'   => '^/(_(profiler|wdt)|css|images|js)/',
+                    'security'  => false,
+                ),
+                'main' => array(
+                    'anonymous' => null,
+                ),
             ),
-            // ...
         ));
 
-Of course, your users' passwords now need to be encoded with this exact algorithm.
-For hardcoded users, you can use the built-in command:
+A "firewall" is your authentication system: the configuration below it defines
+*how* your users will be able to authenticate (e.g. login form, API token, etc).
 
-.. code-block:: terminal
+Only one firewall is active on each request: Symfony uses the ``pattern`` key
+to find the first match (you can also :doc:`match by host or other things </security/firewall_restriction>`).
+The ``dev`` firewall is really a fake firewall: it just makes sure that you don't
+accidentally block Symfony's dev tools - which live under URLs like ``/_profiler``
+and ``/_wdt``.
 
-    $ php bin/console security:encode-password
+All *real* URLs are handled by the ``main`` firewall (no ``pattern`` key means
+it matches *all* URLs). But this does *not* mean that every URL requires authentication.
+Nope, thanks to the ``anonymous`` key, this firewall *is* accessible anonymously.
```

Comment:
Hmm, I don't know if I understand what you're saying. What "won't happen" exactly? You mean that, if you actually have 0 isGranted calls, then your "firewall" is basically never initialized and so you'll see yourself as "not authenticated" instead of "anonymous" on the WDT?